components/match2/commons/match_group_input.lua

```diff
@@ -813,49 +838,67 @@ function MatchGroupInput.getMapVeto(match, allowedVetoes)
 	return data
 end
 
----@param opponents table[]
+---@param opponents {status: string?, score: any}[]
 ---@return boolean
 function MatchGroupInput.isDraw(opponents)
 	if Logic.isEmpty(opponents) then return true end
-	if Array.any(opponents, function (opponent) return opponent.status ~= 'S' and opponent.status ~= 'D' end) then
+	local function opponentHasFinalStatus(opponent)
+		return opponent.status ~= MatchGroupInput.STATUS.SCORE and opponent.status ~= MatchGroupInput.STATUS.DRAW
+	end
+	if Array.any(opponents, opponentHasFinalStatus) then
 		return false
 	end
+	-- Does all opponents have the same score?
 	return #Array.unique(Array.map(opponents, Operator.property('score'))) == 1
 end
 
 -- Check if any opponent has a none-standard status
----@param opponents table[]
+---@param opponents {status: string?}[]
 ---@return boolean
 function MatchGroupInput.hasSpecialStatus(opponents)
-	return Array.any(opponents, function (opponent) return opponent.status ~= 'S' end)
+	return Array.any(opponents, function (opponent) return opponent.status ~= MatchGroupInput.STATUS.SCORE end)
```

Comment:
do we want it to return true for a nil status?
(annos say it can be nil)

Reply:
Not sure, I think changing the annos and ensuring proper input is the way to go

Reply:
changed anno